Automattic/wp-calypso
# HackWeek: Show a banner when the site has 0 content

#### Changes proposed in this Pull Request

As part of the hackweek we want to add a banner in Stats page when the user has not created any content yet. This code depends on backend changes added by the patch D77480-code

#### Testing instructions

Use calypso live branch or checkout this branch (add/banner-in-stats-when-site-has-0-content)
~~if D77480-code is deployed, or sandbox your public-api before perform this tests.~~

Test Events:
Ensure the following events are being sent to Tracks (An easy way to test this is to have [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) monitoring the two next tests)

1. "calypso_stats_no_content_banner_click"
2. "calypso_stats_no_content_banner_dismiss"
3. "calypso_stats_no_content_banner_view"

Test 1:

1. Have a new site where you have not created any content (page or post)
2. Go to Stats page
3. Check that you see the banner
4. Edit (or create if the blog does not have) a page and repeat the test
5. Check that you don't see the banner
6. Delete the page
7. Repeat request with include_pages=true EG: /sites/203903727/site-has-never-published-post?include_pages=true
8. Check that you see the banner
9. Create a blog post
10. Check that you don't see the banner

Test 2:

1. Have a new site where you have not created any content (page or post)
2. Click on the CTA shown by the banner (Write a post)
3. Check that it takes you to write a post (don't create anything)
4. Go to Stats again
5. Click on the dismissal icon
6. Check reload the page and check that you are no longer presented with the banner

*Banner:
![image](https://user-images.githubusercontent.com/5689927/160103603-a738f0ac-ae82-466b-b46f-77bc965b9959.png)


![image](https://user-images.githubusercontent.com/5689927/157896361-6acdbfab-c74c-4cfb-a51e-9c147007916a.png)
